src/main/sensors/battery.c

```diff
@@ -69,16 +73,25 @@ static void updateBatteryVoltage(void)
     static bool vbatFilterIsInitialised;
 
     // store the battery voltage with some other recent battery voltage readings
-    uint16_t vbatSample = vbatLatestADC = adcGetChannel(ADC_BATTERY);
+    uint16_t vbatSample;
+
+    #ifdef USE_ESC_TELEMETRY
+    if (batteryConfig->batteryMeterType == BATTERY_SENSOR_ESC && isEscTelemetryActive()) {
+        vbatSample = vbatLatest = getEscTelemetryVbat();
+    }
+    else
+    #endif
+    {
+        vbatSample = vbatLatest = batteryAdcToVoltage(adcGetChannel(ADC_BATTERY));
+    }
 
     if (debugMode == DEBUG_BATTERY) debug[0] = vbatSample;
 
     if (!vbatFilterIsInitialised) {
         biquadFilterInitLPF(&vbatFilter, VBATT_LPF_FREQ, 50000); //50HZ Update
         vbatFilterIsInitialised = true;
     }
-    vbatSample = biquadFilterApply(&vbatFilter, vbatSample);
-    vbat = batteryAdcToVoltage(vbatSample);
+    vbat = biquadFilterApply(&vbatFilter, vbatSample);
```

Comment:
I'd change that back to `vbatSample` being a raw ADC reading, and doing the scaling after the filtering, to maximise the efficiency of the filtering.